### ISSUE TITLE
Use of `count` incompatible with limit/skip.

### ISSUE THREAD

**Issue by devinivy** [balderdashy/waterline-sequel#55]
The query for counting ([here](https://github.com/balderdashy/waterline-sequel/blob/d1f1b51c5f6ffdb118275250994607dba140d506/sequel/index.js#L120)) will not be compatible with LIMIT and OFFSET clauses.  SELECTing `COUNT(*)` will return one row, so LIMIT and OFFSET should not be applied to that query.  We should change this from (roughly),
```sql
SELECT COUNT(*) FROM table LIMIT 10 OFFSET 10;
```
to
```sql
SELECT COUNT(*) FROM (SELECT * FROM table LIMIT 10 OFFSET 10) AS tableAlias;
```

May need to think about this more for join queries.

**Comment by particlebanana** [comment]
Is the proposed query with the sub-query any faster than just doing the find and returning the number of rows returned? Basically should we even bother with the sub-query and just detect if a limit or offset was used and if so build a normal find query?

**Comment by particlebanana** [comment]
Although I guess then you have to do some fallback in the adapters.

**Comment by devinivy** [comment]
I would suggest building a normal `find` query, then plopping it in a subquery, as above.  That's equivalent to how how waterline polyfills the adapter `count` method (counting rows from a `find` query in memory).  It's always nice keeping the result set small (in this case, always one row) so that time over the wire is minimal and memory is well-conserved.

**Comment by particlebanana** [comment]
Fixed by #56
